### PR TITLE
Soar 5963 Type Converter Unit Tests

### PR DESCRIPTION
**Pull request by aholland-r7**
### Description

Added unit tests for string_to_object and string_to_float actions. Also noticed errors in the help.md and so fixed.

### Testing

No plugin functionality changes, but all tests run and passed

### CODE REVIEW

File: plugins/type_converter/unit_test/test_string_to_float.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        results = test_action.run(working_params)
+        expected = {"output": 0}
+        self.assertEqual(expected, results)
+
+        working_params = {"input": "3.14159265359"}
+        results = test_action.run(working_params)
+        expected = {"output": 3.14159265359}
+        self.assertEqual(expected, results)
+
+        with self.assertRaises(PluginException):
```
Comment (jmcadams-r7) [inline plugins/type_converter/unit_test/test_string_to_float.py:41]: Split these into a new test. Call it like "test_<action>_negative_tests"

File: plugins/type_converter/unit_test/test_string_to_object.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        results = test_action.run(working_params)
+        expected = {"output": {"rapid7": "value"}}
+        self.assertEqual(expected, results)
+
+        working_params = {"input": '{"name":"Doe", "age":100, "city":"El Dorado"}'}
+        results = test_action.run(working_params)
+        expected = {"output": {"name": "Doe", "age": 100, "city": "El Dorado"}}
+        self.assertEqual(expected, results)
+
+        with self.assertRaises(PluginException):
```
Comment (jmcadams-r7) [inline plugins/type_converter/unit_test/test_string_to_object.py:36]: Split into their own tests.

**Review by jmcadams-r7 — CHANGES_REQUESTED** [review]

File: plugins/type_converter/plugin.spec.yaml [hunk 3]
```diff
@@ -196,7 +196,7 @@ actions:
         description: Type converted input
         type: string
         required: true
-        example: Rapid7 Insightconnect
+        example: 'Rapid7 Insightconnect'
```
Comment (aholland-r7) [inline plugins/type_converter/plugin.spec.yaml:199]:
```suggestion
        example: 'Rapid7 InsightConnect'
```

File: plugins/type_converter/help.md [hunk 4]
```diff
@@ -578,16 +579,15 @@ This action converts a string to an object.
...
 
 ```
 {
-  "input": {
-    "object": ["rapid", "7"],
-    "rapid7": "value"
-  }
+  "input": "{
+  'object':['rapid','7'],
+  'rapid7':'value'}"
```
Comment (aholland-r7) [inline plugins/type_converter/help.md:590]:
```suggestion
  "input": "{'object':['rapid','7'], 'rapid7':'value'}"
```

File: plugins/type_converter/help.md [hunk 5]
```diff
@@ -431,7 +430,9 @@ Example output:
 
 ```
 {
-  "output": "{\"object\": [\"rapid\", \"7\"], \"rapid7\": \"value\"}"
+  "output": '{
+  "object": ["rapid", "7"], 
+  "rapid7": "value"}'
```
Comment (aholland-r7) [inline plugins/type_converter/help.md:435]:
```suggestion
  "output": "{'object': ['rapid', '7'], 'rapid7': 'value'}"
```

File: plugins/type_converter/help.md [hunk 6]
```diff
@@ -408,16 +408,15 @@ This action encodes an object to a string.
...
 
 ```
 {
-  "input": {
-    "object": ["rapid", "7"],
-    "rapid7": "value"
-  }
+  "input": "{
+  'object':['rapid','7'],
+  'rapid7':'value'}"
```
Comment (aholland-r7) [inline plugins/type_converter/help.md:419]:
```suggestion
  "input": {
    "object": ["rapid", "7"],
    "rapid7": "value"
  }
```

File: plugins/type_converter/unit_test/test_string_to_float.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+    def test_string_to_float_many_digits(self):
+        log = logging.getLogger("Test")
+        test_action = StringToFloat()
+        test_action.logger = log
+        working_params = {"input": "3.14159265359"}
+        results = test_action.run(working_params)
+        expected = {"output": 3.14159265359}
+        self.assertEqual(expected, results)
+
```
Comment (aholland-r7) [inline plugins/type_converter/unit_test/test_string_to_float.py:52]: Add a _very_ large number (22 digit)

File: plugins/type_converter/unit_test/test_string_to_object.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+            working_params = {"this is a string"}
+            test_action.run(working_params)
+
+        with self.assertRaises(PluginException):
+            working_params = {"input": '{"object: ["rapid", "7"],"rapid7": "value"}'}
+            test_action.run(working_params)
+
+        with self.assertRaises(PluginException):
+            working_params = {"input": '"object": ["rapid", "7"],"rapid7": "value"'}
+            test_action.run(working_params)
```
Comment (aholland-r7) [inline plugins/type_converter/unit_test/test_string_to_object.py:61]: empty string test, or just a space

**Review by jmcadams-r7 — DISMISSED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Review by mberezin-r7 — APPROVED** [review]
LGTM

**Pull request merged**